src/ec/curve25519/x25519.rs

```diff
@@ -69,33 +68,39 @@ fn x25519_public_from_private(public_out: &mut [u8],
 fn x25519_ecdh(out: &mut [u8], my_private_key: &ec::PrivateKey,
                peer_public_key: untrusted::Input)
                -> Result<(), error::Unspecified> {
-    let out =
-        try!(slice_as_array_ref_mut!(out, X25519_ELEM_SCALAR_PUBLIC_KEY_LEN));
+    let out = try!(slice_as_array_ref_mut!(out, SHARED_SECRET_LEN));
+
     // XXX: This shouldn't require dynamic checks, but rustc can't slice an
     // array reference to a shorter array reference. TODO(perf): Fix this.
     let my_private_key =
-        try!(slice_as_array_ref!(
-                &my_private_key.bytes[..X25519_ELEM_SCALAR_PUBLIC_KEY_LEN],
-                X25519_ELEM_SCALAR_PUBLIC_KEY_LEN));
+        try!(slice_as_array_ref!(&my_private_key.bytes[..PRIVATE_KEY_LEN],
+                                 PRIVATE_KEY_LEN));
     let peer_public_key =
         try!(slice_as_array_ref!(peer_public_key.as_slice_less_safe(),
-                                 X25519_ELEM_SCALAR_PUBLIC_KEY_LEN));
+                                 PUBLIC_KEY_LEN));
     bssl::map_result(unsafe {
         GFp_x25519_ecdh(out, my_private_key, peer_public_key)
     })
 }
 
+const ELEM_AND_SCALAR_LEN: usize = ops::ELEM_LEN;
+
+type PrivateKey = [u8; PRIVATE_KEY_LEN];
+const PRIVATE_KEY_LEN: usize = ELEM_AND_SCALAR_LEN;
```

Comment:
Hmm. Well, we got rid of the type alias either way, but the constant remains.

I think, even though we're in a 25519 setting, the naming situation is a little bit different from the Ed25519 case. Here, we aren't holding on to any derived values, and there is no notion of "seed" (since there's no hashing) or explicit "private scalar" in the language of RFC 7748. In my mind, as long as we keep the `private_*` concepts scoped to the relevant protocol modules, it isn't ambiguous— the ambiguity is particular to Ed25519.

In the X25519 case, I don't know what else you would call it— maybe `private_bytes`? That would be accurate, since it isn't a "seed" (we don't hash it), and it isn't sure to be a "scalar" (until masked).

I'm inclined to keep it as-is. What do you think?